### PR TITLE
Migrate Twitter component to Angular

### PR DESCRIPTION
**Pull request by ezequielc**
## Description
Migrate Twitter component to Angular

## Motivation and Context
Angular migration

## How Has This Been Tested?
Locally and apps-beta

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

**Comment by ezequielc** [comment]
@alex-deaconu I validated with authenticated and unauthenticated companies. But the actual authentication flow is broken on staging and it looks like it is from our authenticator side, so I am not going to work on it now. 
Apps Beta authentication flow also did not work but the reason is that apps-beta subdomain is not added to our oauth.io account. I asked Stu to add it so hopefully we can test it tomorrow before releasing this. Please review. Thanks!

**Comment by ezequielc** [comment]
@alex-deaconu Quick update on this, I manage to add apps-beta to the provider and it is working now. Thanks

File: src/app/template-editor/template-components/twitter/twitter.component.ts [hunk 1]
```diff
@@ -1,17 +1,138 @@
...
+    this.attributeDataFactory.setAttributeData(this.componentId, 'isStaging', this.isStaging);
+
+    if (this._validateUsername(this.username)) {
+      this.attributeDataFactory.setAttributeData(this.componentId, 'username', this.username.replace('@', ''));
+    }
+
+    if (this._validateMaxitems(this.maxitems)) {
+      this.attributeDataFactory.setAttributeData(this.componentId, 'maxitems', Number(this.maxitems));
+    }
+    this.changeDetectorRef.detectChanges();
```
Comment (alex-deaconu) [inline src/app/template-editor/template-components/twitter/twitter.component.ts:78]: @ezequielc curious why do we need to detectChanges here. I'm assuming the one above is for the spinner. Thanks.

**Review by alex-deaconu — APPROVED** [review]
@ezequielc looks good 👍 . One question inline. Thanks.

File: src/app/template-editor/template-components/twitter/twitter.component.ts (hunk 1 above)
Comment (ezequielc) [reply src/app/template-editor/template-components/twitter/twitter.component.ts:78]: @alex-deaconu The username validation error was taking a little long to appear. Thanks

Comment (alex-deaconu) [reply src/app/template-editor/template-components/twitter/twitter.component.ts:78]: @ezequielc makes sense, missed that. Thanks!

**Pull request merged**